core/src/com/unciv/UncivGame.kt

```diff
@@ -155,18 +156,30 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
         if (gameInfo.civilizations.count { it.playerType == PlayerType.Human } > 1 && !gameInfo.gameParameters.isOnlineMultiplayer)
             setScreen(PlayerReadyScreen(gameInfo, gameInfo.getPlayerToViewAs()))
         else {
-            worldScreen = WorldScreen(gameInfo, gameInfo.getPlayerToViewAs())
-            setWorldScreen()
+            resetToWorldScreen(WorldScreen(gameInfo, gameInfo.getPlayerToViewAs()))
         }
     }
 
     fun setScreen(screen: BaseScreen) {
+        val oldScreen = getScreen()
         Gdx.input.inputProcessor = screen.stage
         super.setScreen(screen)
+        if (oldScreen != getWorldScreenOrNull()) oldScreen.dispose()
     }
 
-    fun setWorldScreen() {
-        if (screen != null && screen != worldScreen) screen.dispose()
+    /**
+     * If called with null [newWorldScreen], disposes of the current screen and sets it to the current stored world screen. 
+     * If the current screen is already the world screen, the only thing that happens is that the world screen updates.
+     */
+    fun resetToWorldScreen(newWorldScreen: WorldScreen? = null) {
+        if (newWorldScreen != null) {
+            val oldWorldScreen = getWorldScreenOrNull()
+            worldScreen = newWorldScreen
+            // setScreen disposes the current screen, but the old world screen is not the current screen, so need to dispose here
+            if (screen != oldWorldScreen) {
```

Comment:
I think the != is guaranteed to be true?

Reply:
All right. I just looked at the diff and saw all calls either passing null or a freshly constructed instance. If you thought it through who am I to ... wait ... we're not arguing. I'm just lazily prodding to see _if_ you thought that through.